R/calc_views.R

```diff
@@ -25,11 +25,12 @@ calc_views <- function(gsplot){
 }
 
 views <- function(gsplot){
-  gsplot[names(gsplot) %in% 'view']
+  gsplot[grep('view.', names(gsplot))]
 }
 
 non_views <- function(gsplot){
-  gsplot[!names(gsplot) %in% 'view']
+  gsplot[grep('view.', names(gsplot))] <- NULL
```

Comment:
I'd feel more comfortable here if the argument was copied to an intermediate variable.  Manipulating an argument feels dirty.